### DISCUSSION TITLE
Boot options for SPL?

### DISCUSSION THREAD

**Discussion by ltrsunny** [linux-chenxing/linux-chenxing.org#54] [General]
I added some features like MMC and partition support in SPL, then
```
Fixing spl/u-boot-spl.bin
adding 1 pad bytes
Traceback (most recent call last):
  File "/home/dev/mstar-uboot/arch/arm/mach-mstar/fix_ipl_hdr.py", line 71, in <module>
    fix(args.i, args.o)
  File "/home/dev/mstar-uboot/arch/arm/mach-mstar/fix_ipl_hdr.py", line 62, in fix
    write_output(data, chksum, output)
  File "/home/dev/mstar-uboot/arch/arm/mach-mstar/fix_ipl_hdr.py", line 48, in write_output
    o.write(struct.pack(LE16FMT, len(data)))
struct.error: ushort format requires 0 <= number <= (0x7fff * 2 + 1)
```

**Comment by fifteenhex** [comment]
Looks like it's way too big and overflowing the format in the python code.
The SPL needs to be under 49152 bytes. You'll probably need to get rid of support for all other boot methods and use a fixed offset into the SD card to boot from it.

**Reply by ltrsunny** [reply]
> Looks like it's way too big and overflowing the format in the python code. The SPL needs to be under 49152 bytes. You'll probably need to get rid of support for all other boot methods and use a fixed offset into the SD card to boot from it.

The compilation succeeds after I enabled `CONFIG_SPL_MMC_TINY` but it is not trying to load from SD
```

Load IPL_CUST from SPINAND
QUAD MODE ENABLE
 _MDrv_SPINAND_CHECK_ECCSTATUS error
 398
[I]d7
     jump to next partition, block: 0x13
Checksum OK

IPL_CUST g76cd9eb
runUBOOT()
runUBOOT()
[SPINAND]
FORESEE CLKSPI 108M
Verify UBOOT header failed!
Load Backup UBOOT
Verify UBOOT header failed!
Load UBOOT header fails
Disable MMU and D-cache
[HALT]
```

**Comment by ltrsunny** [comment]
And, I noticed that ` CONFIG_SPL_UBI` is force enabled by `TARGET_SOM2D01 [=y] && <choice> && MSTAR_INFINITY2M [=y]`
I think it is better to make it changeable.
